frontends/parsers/p4/p4parser.ypp

```diff
@@ -486,7 +486,7 @@ simpleKeysetExpression
     | expression "&&&" expression { $$ = new IR::Mask(@1 + @3, $1, $3); }
     | expression ".." expression  { $$ = new IR::Range(@1 + @3, $1, $3); }
     | DEFAULT                     { $$ = new IR::DefaultExpression(@1); }
-    | "_"                         { $$ = new IR::DefaultExpression(@1); }
+    | DONTCARE                    { $$ = new IR::DefaultExpression(@1); }
```

Comment:
This is an unnecessary obfuscation -- its generally better to use the literals for textual tokens (punctuation and the like) rather than names, as this makes it look like DONTCARE is a keyword `dontcare` rather than a single underscore.

Reply:
In this case we should get rid of the DONTCARE terminal altogether.

Reply:
The problem is that a keyword is needed for the lexer to generate the token as flex (or more precisely the C++ interface between flex and bison) does not support referring to tokens by string literal.  So for all the punctuation-type tokens, we need to define an alias identifier that is used by flex to create the token object.